GSA-TTS/FAC
# 3696 address historical awards with repeated references

## Description

Some historical reports failed migration due to repeated reference numbers in the findings section. Our current validation rule does not allow repeated reference numbers in findings workbook. The team has decided to migrate these reports as is by skipping that validation rule for those specific reports. This PR addresses that issue. The solution involves:

- Adding logic to look for duplicate references in findings records before the creation of the findings workbook in the migration module.
- When a duplicate is detected, tracking the bad records and setting a global flag that will allow the report to bypass the corresponding validation rule once it reaches that logic.
- If the report does not break any other validation rule, saving the tracked bad records into the InvalidAuditRecord table and updating the ReportMigrationStatus table; otherwise, clearing the track.
- Clear the flag

## To test:

1. Launch the stack from the main branch.
2. Attempt to migrate the audit report with `dbkey = 245186` and `audit_year = 2016` and notice the migration failure.
3. Switch to this PR branch and re-run the migration for the same report and observe that it processes properly.
4. Check the InvalidAuditRecord table and ReportMigrationStatus table for records related to this audit.


## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
